lib/smart_proxy_salt/salt.rb

```diff
@@ -14,10 +14,19 @@ class Plugin < ::Proxy::Plugin
 
       default_settings :autosign_file      => '/etc/salt/autosign.conf',
                        :salt_command_user  => 'root',
-                       :use_api            => false
+                       :use_api            => false,
+                       :saltfile           => '/etc/foreman-proxy/settings.d/salt.saltfile'
 
       http_rackup_path File.expand_path('salt_http_config.ru', File.expand_path('../', __FILE__))
       https_rackup_path File.expand_path('salt_http_config.ru', File.expand_path('../', __FILE__))
+
+      after_activation do
+        require 'smart_proxy_dynflow'
+        require 'smart_proxy_salt/salt_runner'
+        require 'smart_proxy_salt/salt_task_launcher'
+
+        Proxy::Dynflow::TaskLauncherRegistry.register('salt', SaltTaskLauncher)
+      end
```

Comment:
This does trigger the legacy module loader. I wonder if we can implement this using `load_classes` and `wire_up_dependencies`.